tsdb/tsm1/engine_delete_prefix.go

```diff
@@ -77,24 +77,30 @@ func (e *Engine) DeleteBucketRange(name []byte, min, max int64) error {
 
 	// ApplySerialEntryFn cannot return an error in this invocation.
 	_ = e.Cache.ApplyEntryFn(func(k []byte, _ *entry) error {
-		if bytes.HasPrefix(k, name) {
-			if deleteKeys == nil {
-				deleteKeys = make([][]byte, 0, 10000)
-			}
-			deleteKeys = append(deleteKeys, k)
+		if !bytes.HasPrefix(k, name) {
+			return nil
+		}
+		if pred != nil && !pred.Matches(k) {
+			return nil
+		}
 
-			// we have to double check every key in the cache because maybe
-			// it exists in the index but not yet on disk.
-			possiblyDead.keys[string(k)] = struct{}{}
+		if deleteKeys == nil {
+			deleteKeys = make([][]byte, 0, 10000)
```

Comment:
It's not. I think this is leftover from a long porting ago. I'm actually happy removing it entirely and just letting append find the right size, rather than hinting (and possibly doing a ton of extra work allocating zeroed memory that contains pointers, etc.)